tasks/run.rb

```diff
@@ -99,6 +99,13 @@ def get_result_from_report(last_run_report, run_result, start_time)
         obj.tag = nil if obj.respond_to?(:tag=)
       end
 
+      # check if the run is marked as failed
+      if report.to_ruby['status'] == 'failed'
+        return error_result(
+          'puppet_agent/agent-run-error',
```

Comment:
Are we free to add new `error_type`s? Is there a schema for error types? Or is there some translation that happens to map this to a human readable version?

Reply:
I've really no idea :( My impression is that we can just add random text here: https://www.puppet.com/docs/bolt/latest/writing_tasks#returning-errors-in-tasks